lib/AST/ASTImporter.cpp

```diff
@@ -2559,7 +2545,18 @@ Decl *ASTNodeImporter::VisitFunctionDecl(FunctionDecl *D) {
     }
 
   // Import the body, if any.
-  if (Stmt *FromBody = D->getBody()) {
+  const FunctionDecl* FromDefinition = nullptr;
+  if (Stmt *FromBody = D->getBody(FromDefinition)) {
+
+    // If we have the defintion in the FromCtx, mark that as imported.
+    // This handles difficulties when we have a recursive function with a
+    // prototype and a definition in the same ToCtx.  E.g. importing the
+    // prototype below would import the definition twice without this.
+    // "void f(); void f() { f(); }"
+    // This is because in the definition the CallExpression refers to the
+    // definition, not to the prototype.
```

Comment:
This check should be inserted:
```
    if (!Importer.GetAlreadyImportedOrNull(
          const_cast<FunctionDecl*>(FromDefinition)))
```

Reply:
Oh, as far as I can see we noticed this almost the same time :)